packages/flutter/lib/src/widgets/focus_traversal.dart

```diff
@@ -146,15 +212,162 @@ abstract class FocusTraversalPolicy {
-  node.requestFocus();
-  Scrollable.ensureVisible(node.context, alignment: 1.0, alignmentPolicy: alignmentPolicy);
+  /// Sorts the given `descendants` into focus order.
+  ///
+  /// Subclasses should override this to implement a different sort for [next]
+  /// and [previous] to use in their ordering. If the returned iterable omits a
+  /// node that is a descendant of the given scope, then the user will be unable
+  /// to use next/previous keyboard traversal to reach that node, and if that
+  /// node is used as the originator of a call to next/previous (i.e. supplied
+  /// as the argument to [next] or [previous]), then the next or previous node
+  /// will not be able to be determined and the focus will not change.
+  ///
+  /// This is not used for directional focus ([inDirection]), only for
+  /// determining the focus order for [next] and [previous].
+  ///
+  /// When implementing an override for this function, be sure to use
+  /// [mergeSort] instead of Dart's default list sorting algorithm when sorting
+  /// items, since the default algorithm is not stable (items deemed to be equal
+  /// can appear in arbitrary order, and change positions between sorts), whereas
+  /// [mergeSort] is stable.
+  @protected
+  Iterable<FocusNode> sortDescendants(Iterable<FocusNode> descendants);
+
+  _FocusTraversalGroupMarker _getMarker(BuildContext context) {
+    return context?.getElementForInheritedWidgetOfExactType<_FocusTraversalGroupMarker>()?.widget as _FocusTraversalGroupMarker;
+  }
+
+  // Sort all descendants, taking into account the FocusTraversalGroup
+  // that they are each in, and filtering out non-traversable/focusable nodes.
+  List<FocusNode> _sortAllDescendants(FocusScopeNode scope) {
+    assert(scope != null);
+    final _FocusTraversalGroupMarker scopeGroupMarker = _getMarker(scope.context);
+    final FocusTraversalPolicy defaultPolicy = scopeGroupMarker?.policy ?? ReadingOrderTraversalPolicy();
+    // Build the sorting data structure, separating descendants into groups.
+    final Map<FocusNode, _FocusTraversalGroupInfo> groups = <FocusNode, _FocusTraversalGroupInfo>{};
+    for (final FocusNode node in scope.descendants) {
+      final _FocusTraversalGroupMarker groupMarker = _getMarker(node.context);
+      final FocusNode groupNode = groupMarker?.focusNode;
+      // Group nodes need to be added to their parent's node, or to the "null"
+      // node if no parent is found. This creates the hierarchy of group nodes
+      // and makes it so the entire group is sorted along with the other members
+      // of the parent group.
+      if (node == groupNode) {
+        // To find the parent of the group node, we need to skip over the parent
+        // of the Focus node in _FocusTraversalGroupState.build, and start
+        // looking with that node's parent, since
+        // getElementForInheritedWidgetOfExactType will return the context it
```

Comment:
Update this comment since getElementForInheritedWidgetOfExactType is no longer called here?